### PR TITLE
Rewind_Exception test and Advance_Exception test

### PR DESCRIPTION
**Pull request by lchesson**
Added test for Rewind and Advance in System.Memory.Performance to test Exceptions when count is less then 0.

### CODE REVIEW

**Comment by lchesson** [comment]
completed while pairing with @amccombs20

File: src/System.Memory/tests/SequenceReader/Advance.cs [hunk 1]
```diff
@@ -57,5 +57,18 @@ public void PastEmptySegments()
...
+        public void Advance_Exception()
+        {
+            ReadOnlySequence<byte> bytes = SequenceFactory.Create(new byte[][] {
+                new byte[] { 0          },
+                new byte[] { 1, 2       },
+                new byte[] { 3, 4       },
+                new byte[] { 5, 6, 7, 8 }
+            });
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => new SequenceReader<byte>(bytes).Advance(-1));
```
Comment (ahsonkhan) [inline src/System.Memory/tests/SequenceReader/Advance.cs]: Can you also add an assert for advancing beyond bytes.Length (and not just < 0)?

File: src/System.Memory/tests/SequenceReader/Rewind.cs [hunk 2]
```diff
@@ -67,5 +67,18 @@ public void Rewind_ByOne()
...
+        public void Rewind_Exception()
+        {
+            ReadOnlySequence<byte> bytes = SequenceFactory.Create(new byte[][] {
+                new byte[] { 0          },
+                new byte[] { 1, 2       },
+                new byte[] { 3, 4       },
+                new byte[] { 5, 6, 7, 8 }
+            });
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => new SequenceReader<byte>(bytes).Rewind(-1));
```
Comment (ahsonkhan) [inline src/System.Memory/tests/SequenceReader/Rewind.cs]: Similarly, can you add an assert where you advance by some value x, and then rewind by x + 1 (which should result in OutOfRange Exception as well).

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

**Comment by danmosemsft** [comment]
@ahsonkhan can this merge now?

**Pull request merged**